### ISSUE TITLE
Filter for Platform (mobile, web or desktop)?

### ISSUE THREAD

**Issue by fabioselau077** [dart-lang/pub-dev#5126]
Hi, with the growing expansion of Flutter Web I feel the need to filter search packages by platform.
For example, I search for "menu" and in the filters select Flutter Web, then all packages that are compatible with Flutter Web are returned.

**Comment by isoos** [comment]
@fabioselau077: we do have filters for that, e.g. on the following page: https://pub.dev/flutter/packages?q=&platform=web
Is that what you were looking for?

We are also planning to update the UI of these filters to make them more discoverable...

**Comment by Driky** [comment]
I feel we should be able to search package that works `only` for the filtered platforms. In my case that would like to be able to look for packages/plugins specific to `windows`, `macos` and `linux`. That would make discovering desktop packages and plugin easier.

**Comment by jonasfj** [comment]
I don't understand the use-case? You want to not see plugins that supports multiple platforms?

Maybe what you want is the ability to only see packages that are in fact plugins?

**Comment by Driky** [comment]
@jonasfj yes being able to see plugin that only support the specified platform would help find plugins for OS specific functionalities. And yes gain, being able to only see plugin would be a nice addition.

**Comment by jonasfj** [comment]
@Driky this seems a little backwards..

Let's imagine there are two plugins for blocking screen locking:
 * `package:lock_screen` which supports android, ios and web,
 * `package:iscreen_lock` which only supports ios

Why would you not want to discover `package:lock_screen`? It might be better maintained, clearly the authors cared enough to add support for multiple platforms.
